packages/server/src/fhir/repo.ts

```diff
@@ -2564,3 +2571,15 @@ function patchObject(obj: any, patch: Operation[]): void {
     throw new OperationOutcomeError(normalizeOperationOutcome(err));
   }
 }
+
+/**
+ * Apply a maximum string length to ensure the value can accommodate the maximum
+ * size for a btree index entry: 2704 bytes. Be as conservative as possible to avoid
+ * write errors; so limit to 675 characters to accommodate the entire string being
+ * random 4-byte UTF-8 characters.
+ * @param value - The column value to truncate.
+ * @returns The possibly truncated column value.
+ */
+function truncateTextColumn(value: string): string {
+  return value.substring(0, 675);
```

Comment:
Is this not a breaking change for folks who currently have resources with strings that are longer than this, but under the byte limit?

Reply:
Yes, there would be breaking changes at search time for those resources. The most likely would be a `:contains` with a query in the latter part of the string.

Reply:
Discussed offline.  Recommend using `Blob(str).size` rather than simple string length.  (Although, we still need to figure out how to truncate in a safe way, so we don't cut into a multibyte code point or whatever)

For strings that are too long for the btree index, then it won't be a breaking change, because the initial write would have been a database error in the first place.

For strings that do fit in the btree index, this shouldn't be a change.